transport/response.go

```diff
@@ -32,6 +32,11 @@ type Response struct {
 type ResponseWriter interface {
 	io.Writer
 
+	// Options for the request/response that this ResponseWriter is attached to.
+	Options() Options
+	// TODO(abg) Options should not be attached to the ResponseWriter. Instead
+	// they should be part of the Handle interface.
```

Comment:
It would affect the readability of this PR because it touches more files. I have the change done locally, but I'll create a separate PR for that -- probably against this one.